### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by andyjansson**
This rewrite is done with the intent of fixing inconsistent behaviour between reduce-css-calc and the W3C specification, as well as enabling better optimisation of `calc()` expressions. The rewrite is done from the ground-up in ES6 while still staying compatible with the existing API. It reduces `calc()` functions by transforming ASTs generated from the expressions.  

The rewrite also solves the following issues:

* moOx/reduce-css-calc#3
* moOx/reduce-css-calc#2
* moOx/reduce-css-calc#1
* postcss/postcss-calc#33
* postcss/postcss-calc#25
* ben-eb/cssnano#320
* ben-eb/cssnano#319
* ben-eb/cssnano#211

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "presets": ["es2015-loose", "stage-0"],
```
Comment (MoOx) [inline .babelrc]: why not "just" babel-preset-latest?

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
```
Comment (MoOx) [inline .eslintrc:1]: Any preset maybe?

File: package.json [hunk 3]
```diff
@@ -7,27 +7,40 @@
...
     "calc"
   ],
-  "author": "Maxime Thirouin",
-  "license": "MIT",
-  "repository": "https://github.com/MoOx/reduce-css-calc.git",
+  "main": "dist/index.js",
   "files": [
-    "index.js"
+    "dist",
+    "LICENSE"
```
Comment (MoOx) [inline package.json]: LICENSE is unecessary and automatic like README

**Review by MoOx — CHANGES_REQUESTED** [review]
not sure it's a good idea to have postcss as a dep for a non postcss project... (I had in mind that other stuff like rework might use it).
If that's too postcss related why not merging this with postcss-calc directly?

Also it seems you don't cover all previous tests with the new tests so regressions are possible. Any reason to not keep previous tests in addition to new ones?

**Comment by andyjansson** [comment]
@MoOx it's not dependent upon postcss. It might just be some leftover copypasta from the postcss-calc rewrite. I'll fix it when I get home from work.

File: .eslintrc (hunk 2 above)
Comment (ben-eb) [reply .eslintrc:1]: @MoOx https://github.com/MoOx/eslint-config-i-am-meticulous ? 😄

Comment (MoOx) [reply .eslintrc:1]: Why not

**Comment by ben-eb** [comment]
https://github.com/MoOx/reduce-css-calc/issues/22 supersedes this pull request.

**Pull request closed without merging**